pkg/sentry/kernel/fd_table.go

```diff
@@ -219,18 +222,26 @@ func (f *FDTable) NewFDs(ctx context.Context, fd int32, files []*fs.File, flags
 			end = int32(lim.Cur)
 		}
 		if fd >= end {
+			// FD exceeds limit, no available FDs until old FDs get released.
+			f.next = end
```

Comment:
1. f.next is changed without locking f.mu. This doesn't look right, does it?
2. I don't understand why we need to change f.next in this case.

Reply:
1. You're right.
2. My misunderstandings.:joy: